linera-base/src/crypto/ed25519.rs

```diff
@@ -140,56 +142,47 @@ impl<'de> Deserialize<'de> for Signature {
     }
 }
 
-impl fmt::Display for Signature {
+impl fmt::Display for Ed25519Signature {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         let s = hex::encode(self.0.to_bytes());
         write!(f, "{}", s)
     }
 }
 
-impl fmt::Debug for Signature {
+impl fmt::Debug for Ed25519Signature {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         write!(f, "{}", hex::encode(&self.0.to_bytes()[0..8]))
     }
 }
 
-doc_scalar!(Signature, "A signature value");
+doc_scalar!(Ed25519Signature, "An Ed25519 signature value");
 
-/// A BCS-signable struct for testing.
 #[cfg(with_testing)]
-#[derive(Debug, Serialize, Deserialize)]
-pub struct TestString(pub String);
+mod ed25519_tests {
```

Comment:
The parent module is already called `ed25519`. It's common to just call the tests module `tests`.